### PR TITLE
Add postgresql schema support.

### PR DESCRIPTION
**Pull request by angryfoxx**
## Checklist

Please update this checklist as you complete each item:

- [x] Tests have been developed for bug fixes or new functionality.
- [x] The changelog has been updated, if necessary.
- [x] Documentation has been updated, if necessary.
- [ ] GitHub Issues closed by this PR have been linked.

<sub>By submitting this pull request I agree that all contributions comply with this project's open source license(s).</sub>

### CODE REVIEW

**Comment by angryfoxx** [comment]
I am not sure should I update the changelog. @Archmonger what do you think ?

File: dbbackup/management/commands/dbbackup.py [hunk 1]
```diff
@@ -49,7 +49,10 @@ class Command(BaseDbBackupCommand):
             help="Encrypt the backup files",
         ),
         make_option(
-            "-o", "--output-filename", default=None, help="Specify filename on storage"
+            "-o",
+            "--output-filename",
+            default=None,
+            help="Specify filename on storage",
```
Comment (Archmonger) [inline dbbackup/management/commands/dbbackup.py:55]: Please revert unneeded formatting changes

**Review by Archmonger — CHANGES_REQUESTED** [review]

**Comment by Archmonger** [comment]
Yes. Since this is a feature change, you'll need a changelog bullet point.

File: dbbackup/management/commands/dbbackup.py (hunk 1 above)
Comment (angryfoxx) [reply dbbackup/management/commands/dbbackup.py:55]: ofc

File: dbbackup/tests/commands/test_dbrestore.py [hunk 2]
```diff
@@ -46,6 +46,8 @@ def setUp(self):
         self.command.input_database_name = None
         self.command.database_name = "default"
         self.command.connector = get_connector("default")
+        self.command.schemas = []
+        self.command.no_owner = False
```
[the remark this answers is not in the document]
Comment (angryfoxx) [reply dbbackup/tests/commands/test_dbrestore.py:50]: yes I missed it sry

File: dbbackup/db/postgresql.py [hunk 3]
```diff
@@ -57,9 +65,14 @@ def _restore_dump(self, dump):
...
         cmd += " --set ON_ERROR_STOP=on"
+
+        if self.schemas:
+            cmd += " -n " + " -n ".join(self.schemas)
+
         if self.single_transaction:
             cmd += " --single-transaction"
-        cmd += " {}".format(self.settings["NAME"])
+
+        cmd += f" {self.settings['NAME']}"
```
Comment (Archmonger) [inline dbbackup/db/postgresql.py:75]: There's several unnecessary f-string changes in this PR.

There's a ton of small changes like this that need to happen across the whole repo. Will leave that for a future refactoring PR.

File: dbbackup/management/commands/dbbackup.py (hunk 1 above)
Comment (angryfoxx) [reply dbbackup/management/commands/dbbackup.py]: fixed @Archmonger

File: docs/databases.rst [hunk 4]
```diff
@@ -198,6 +198,15 @@ ADMIN_PASSWORD
...
 example.
 
+SCHEMAS
+~~~~~~~
+
+Schemas to specify when dumping database.
+Dump only schemas matching pattern; this selects both the schema itself, and all its contained objects.
+When this option is not specified, all non-system schemas in the target database will be dumped.
+Multiple schemas can be selected.
+This option only available for postgresql connectors.
```
[the remark this answers is not in the document]
Comment (angryfoxx) [reply docs/databases.rst]: Is it better now @Archmonger

File: dbbackup/tests/commands/test_dbrestore.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (angryfoxx) [reply dbbackup/tests/commands/test_dbrestore.py]: fixed

File: dbbackup/management/commands/dbbackup.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
 
 
 class Command(BaseDbBackupCommand):
-    help = "Backup a database, encrypt and/or compress and write to " "storage." ""
+    help = "Backup a database, encrypt and/or compress and write to 'storage.'"
```
Comment (Archmonger) [inline dbbackup/management/commands/dbbackup.py:14]: to be honest there's no reason for `'storage'` to be in quotes here. Let's just remove that.

**Review by Archmonger — CHANGES_REQUESTED** [review]

File: dbbackup/db/postgresql.py (hunk 3 above)
Comment (angryfoxx) [reply dbbackup/db/postgresql.py]: @Archmonger fixed.

File: dbbackup/management/commands/dbbackup.py (hunk 5 above)
Comment (angryfoxx) [reply dbbackup/management/commands/dbbackup.py]: is it better rn? @Archmonger

File: dbbackup/management/commands/dbbackup.py [hunk 6]
```diff
@@ -60,6 +60,13 @@ class Command(BaseDbBackupCommand):
         make_option(
             "-x", "--exclude-tables", default=None, help="Exclude tables from backup"
         ),
+        make_option(
+            "-n",
+            "--schema",
+            action="append",
+            default=[],
+            help="Specify schema(s) to backup. Can be used multiple times.",
```
Comment (Archmonger) [inline dbbackup/management/commands/dbbackup.py:68]: I have a general question, is there an actual use-case of using multiple schemas for one backup session?

File: dbbackup/management/commands/dbbackup.py [hunk 7]
```diff
@@ -103,22 +111,32 @@ def _save_new_backup(self, database):
...
             outputfile = compressed_file
+
         if self.encrypt:
             encrypted_file, filename = utils.encrypt_file(outputfile, filename)
             outputfile = encrypted_file
+
         # Set file name
         filename = self.filename or filename
-        self.logger.debug("Backup size: %s", utils.handle_size(outputfile))
+        self.logger.info("Backup tempfile created: %s", utils.handle_size(outputfile))
```
Comment (Archmonger) [inline dbbackup/management/commands/dbbackup.py:133]: I'd rather this say a debug log

File: docs/changelog.rst [hunk 8]
```diff
@@ -7,6 +7,10 @@ Unreleased
 * Fix restore of database from S3 storage by reintroducing inputfile.seek(0) to utils.uncompress_file
 * Fix bug where dbbackup management command would not respect settings.py:DBBACKUP_DATABASES
 
+4.2.0 (2022-01-30)
+------------------
+* Add PostgreSQL Schema support by @angryfoxx in https://github.com/jazzband/django-dbbackup/pull/507
```
Comment (Archmonger) [inline docs/changelog.rst:12]: Move this to the `unreleased` section above

**Comment by Archmonger** [comment]
This can be merged after handling the two remaining comments.

File: dbbackup/management/commands/dbbackup.py (hunk 6 above)
Comment (angryfoxx) [reply dbbackup/management/commands/dbbackup.py:68]: I haven't encountered before but Postgres supports it.

File: dbbackup/management/commands/dbbackup.py (hunk 7 above)
Comment (angryfoxx) [reply dbbackup/management/commands/dbbackup.py]: fixed

File: docs/changelog.rst (hunk 8 above)
Comment (angryfoxx) [reply docs/changelog.rst]: fixed

**Comment by angryfoxx** [comment]
> This can be merged after handling the two remaining comments.
> 
> EDIT: Actually there are four coverage hits that should be fixed first. See [above](https://github.com/jazzband/django-dbbackup/pull/507#issuecomment-1916303341).

@Archmonger  The Schemas only available for Postgres so. I need to patch the database settings while testing. But I failed, could you give an example of how to do it

**Comment by Archmonger** [comment]
Are you referring to the settings.py values? If so, [this PR](https://github.com/jazzband/django-dbbackup/pull/484/files#diff-65fbb529332ccf542762384e10edc113ee8973a404b1b99e42dee1cda71829a8) shows a good example of how to do that via `@patch`

**Comment by angryfoxx** [comment]
Hi @Archmonger, Could you review it again? I have created schema test for dbrestore just before and I think everything looks good.

**Review by Archmonger — APPROVED** [review]
Looks good to me.

**Pull request merged**